packages/driver/src/cy/multi-domain/validator.ts

```diff
@@ -12,7 +13,7 @@ export class Validator {
   }
 
   validate ({ callbackFn, data, domain, done, doneReference }) {
-    if (typeof domain !== 'string') {
+    if (typeof domain !== 'string' || !(isValidDomain(domain, { allowUnicode: true, subdomain: false }))) {
```

Comment:
I was going to suggest maybe using `_.isString` here to account for the string constructor, like `new String('foobar.com')`, but turns out `is-valid-domain` doesn't account for that and actually does the check the string conditional we are doing [here](https://github.com/miguelmota/is-valid-domain/blob/master/index.js#L6). We might be able to remove the first conditional.
```suggestion
    if !(isValidDomain(domain, { allowUnicode: true, subdomain: false })) {
```